File: java project/project/utils/MenuPrinter.java
package project.utils;

/**
 * Prints neat ASCII menus and messages for the terminal application.
 * Keeping display code here makes the service classes easier to read.
 */
public class MenuPrinter {

    /**
     * Prints the application welcome screen.
     */
    public static void printWelcomeScreen() {
        System.out.println("==========================================");
        System.out.println("      PUBLIC GRIEVANCE SORTING SYSTEM");
        System.out.println("==========================================");
        System.out.println();
        System.out.println("1. User Login");
        System.out.println("2. User Signup");
        System.out.println("3. Admin Login");
        System.out.println("4. Exit");
        System.out.println();
    }

    /**
     * Prints the user dashboard menu.
     */
    public static void printUserDashboard() {
        System.out.println("------------------------------------------");
        System.out.println("               USER DASHBOARD");
        System.out.println("------------------------------------------");
        System.out.println("1. Submit Complaint");
        System.out.println("2. View My Complaints");
        System.out.println("3. Search Complaint by ID");
        System.out.println("4. Edit Pending Complaint");
        System.out.println("5. Delete Pending Complaint");
        System.out.println("6. Profile");
        System.out.println("7. Logout");
        System.out.println("------------------------------------------");
    }

    /**
     * Prints the admin dashboard menu.
     */
    public static void printAdminDashboard() {
        System.out.println("------------------------------------------");
        System.out.println("              ADMIN DASHBOARD");
        System.out.println("------------------------------------------");
        System.out.println("1. View All Complaints");
        System.out.println("2. Search Complaint");
        System.out.println("3. View Pending");
        System.out.println("4. View Accepted");
        System.out.println("5. View Resolved");
        System.out.println("6. View Rejected");
        System.out.println("7. Sort Complaints");
        System.out.println("8. Change Complaint Status");
        System.out.println("9. Give Admin Remark");
        System.out.println("10. Dashboard Statistics");
        System.out.println("11. Logout");
        System.out.println("------------------------------------------");
    }

    /**
     * Prints a short greeting for a logged-in user.
     *
     * @param name user's full name
     */
    public static void printWelcomeBack(String name) {
        System.out.println("------------------------------------");
        System.out.println("Hello, " + name);
        System.out.println("Welcome Back");
        System.out.println("------------------------------------");
    }

    /**
     * Prints a goodbye message.
     */
    public static void printGoodbye() {
        System.out.println();
        System.out.println("==========================================");
        System.out.println("   Thank you for using the system.");
        System.out.println("             Goodbye!");
        System.out.println("==========================================");
    }

    /**
     * Prints a simple loading animation style message.
     *
     * @param message message to show
     */
    public static void printLoading(String message) {
        System.out.println(message + " ...");
    }

    /**
     * Prints a section heading.
     *
     * @param title heading text
     */
    public static void printSection(String title) {
        System.out.println();
        System.out.println("==========================================");
        System.out.println("  " + title);
        System.out.println("==========================================");
    }
}

File: cpp_project/utils/MenuPrinter.cpp
#include "MenuPrinter.h"
#include <iostream>

namespace project {
namespace utils {

void MenuPrinter::printWelcomeScreen() {
    std::cout << "==========================================\n";
    std::cout << "      PUBLIC GRIEVANCE SORTING SYSTEM\n";
    std::cout << "==========================================\n\n";
    std::cout << "1. User Login\n";
    std::cout << "2. User Signup\n";
    std::cout << "3. Admin Login\n";
    std::cout << "4. Exit\n\n";
}

void MenuPrinter::printUserDashboard() {
    std::cout << "------------------------------------------\n";
    std::cout << "               USER DASHBOARD\n";
    std::cout << "------------------------------------------\n";
    std::cout << "1. Submit Complaint\n";
    std::cout << "2. View My Complaints\n";
    std::cout << "3. Search Complaint by ID\n";
    std::cout << "4. Edit Pending Complaint\n";
    std::cout << "5. Delete Pending Complaint\n";
    std::cout << "6. Profile\n";
    std::cout << "7. Logout\n";
    std::cout << "------------------------------------------\n";
}

void MenuPrinter::printAdminDashboard() {
    std::cout << "------------------------------------------\n";
    std::cout << "              ADMIN DASHBOARD\n";
    std::cout << "------------------------------------------\n";
    std::cout << "1. View All Complaints\n";
    std::cout << "2. Search Complaint\n";
    std::cout << "3. View Pending\n";
    std::cout << "4. View Accepted\n";
    std::cout << "5. View Resolved\n";
    std::cout << "6. View Rejected\n";
    std::cout << "7. Sort Complaints\n";
    std::cout << "8. Change Complaint Status\n";
    std::cout << "9. Give Admin Remark\n";
    std::cout << "10. Dashboard Statistics\n";
    std::cout << "11. Logout\n";
    std::cout << "------------------------------------------\n";
}

void MenuPrinter::printWelcomeBack(const std::string& name) {
    std::cout << "------------------------------------\n";
    std::cout << "Hello, " << name << "\n";
    std::cout << "Welcome Back\n";
    std::cout << "------------------------------------\n";
}

void MenuPrinter::printGoodbye() {
    std::cout << "\n==========================================\n";
    std::cout << "   Thank you for using the system.\n";
    std::cout << "             Goodbye!\n";
    std::cout << "==========================================\n";
}

void MenuPrinter::printLoading(const std::string& message) {
    std::cout << message << " ...\n";
}

void MenuPrinter::printSection(const std::string& title) {
    std::cout << "\n==========================================\n";
    std::cout << "  " << title << "\n";
    std::cout << "==========================================\n";
}

}
}
